.github/workflows/test.yaml

```diff
@@ -0,0 +1,125 @@
+name: test
+on: [push, pull_request]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    env:
+      SM_ENVIRONMENT: LOCAL
+      SM_DEV_DB_PROJECT: sm_dev
+      SM_DEV_DB_HOST: '127.0.0.1'
+      SM_DEV_DB_PORT: '3307'
+      SM_DEV_DB_USER: root
+      SM_DEV_DB_PASSWORD: root
+      CONDA: /usr/share/miniconda
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - uses: actions/checkout@main
+
+      - name: gcloud setup
+        uses: google-github-actions/setup-gcloud@master
+        with:
+          project_id: sample-metadata
+          service_account_key: ${{ secrets.GCP_SERVER_DEPLOY_KEY }}
+          export_default_credentials: true
+
+      - name: gcloud docker auth
+        run: |
+          gcloud auth configure-docker australia-southeast1-docker.pkg.dev
```

Comment:
I don't think we need this for the unittesting

Reply:
Will it be able to check secrets without that?

Also happy to use a separate service account rather than the deploy one